src/Misc.pm

```diff
@@ -4631,6 +4634,116 @@ sub closeShop {
 	message T("Shop closed.\n");
 }
 
+sub makeBuyerShop {
+	if ($buyershopstarted) {
+		error T("A shop has already been opened.\n");
+		return;
+	}
+
+	return unless $char;
+
+	my $max_items = 2;
+
+	if (!$char->{skills}{ALL_BUYING_STORE}{lv}) { # don't have skill but have the necessary item
+		my $item = $char->inventory->getByNameID(12548);
+		if(!$item) {
+			error T("You don't have the Buying Store skill or Black Market Bulk Buyer Shop License.\n");
+			return;
+		} else {
+			$item->use;
```

Comment:
another important thing: the code should look if there are items to sell before use the item
because if has no items to sell, char get stuck, because the skill were already casted

this also happens with normal openshop. Something i want to change

Reply:
![image](https://user-images.githubusercontent.com/10372732/42410705-a6372932-81c4-11e8-9689-de6f6bf372e6.png)
fixed